### PR TITLE
refactor utility functions

### PR DESCRIPTION
**Pull request by rlkelly**
## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: src/Tezos/Address.h [hunk 1]
```diff
@@ -23,3 +23,6 @@ class Address {
 struct TWTezosAddress {
     TW::Tezos::Address impl;
 };
+
+int checkDecodeAndDropPrefix(const std::string& input, size_t prefixLength, uint8_t *prefix, uint8_t *output);
```
Comment (keefertaylor) [inline src/Tezos/Address.h]: Shouldn't this live in a common file?

File: tests/Tezos/TWTezosAddressTests.cpp [hunk 2]
```diff
@@ -40,8 +40,8 @@ TEST(TWTezosAddress, TestValidAddresses) {
     ASSERT_TRUE(TWTezosAddressIsValid(tezosAddressData));
     ASSERT_TRUE(TWTezosAddressInitWithString(tezosAddress, tezosAddressString));
 
-    TWData *descriptionData = TWBase58DecodeNoCheck(TWTezosAddressDescription(tezosAddress));
-    ASSERT_TRUE(TWDataEqual(descriptionData, tezosAddressData));
+    // TWData *descriptionData = TWBase58DecodeNoCheck(TWTezosAddressDescription(tezosAddress));
```
Comment (keefertaylor) [inline tests/Tezos/TWTezosAddressTests.cpp]: Revert?

File: tests/Tezos/TransactionTests.cpp [hunk 3]
```diff
@@ -21,7 +20,7 @@ TEST(TezosTransaction, forgePayment) {
     257,
     1,
     Address("tz1Yju7jmmsaUiG9qQLoYv35v5pHgnWoLWbt"),
-    type
+    TransactionType::PAYMENT
```
Comment (keefertaylor) [inline tests/Tezos/TransactionTests.cpp]: Tezos calls these `transactions` and it gets serialized to a field called `kind`. 

I'd expected something like `Kind::TRANSACTION`.

Also, why not just use the enum on the proto?

Comment (rlkelly) [reply tests/Tezos/TransactionTests.cpp]: good point!

File: tests/Tezos/TWTezosAddressTests.cpp (hunk 2 above)
Comment (rlkelly) [reply tests/Tezos/TWTezosAddressTests.cpp]: changed

**Review by keefertaylor — APPROVED** [review]

**Pull request merged**